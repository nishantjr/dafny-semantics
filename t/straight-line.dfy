method straightLine(x: int)
    returns (r: int)
  requires x > 0
  ensures r > 0
{
  r := 2 * x ;
  r := r + 3 ;
  r := r - 2 ;
}
