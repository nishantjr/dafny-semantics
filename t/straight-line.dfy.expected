    {
      Result
    #Equals
      <generatedTop>
        <k>
          .
        </k>
        <methods>
          Main |-> method Main ( .ArgDecls ) returns ( r : int , .ArgDecls ) requires true ensures true { assert Verify ( 0 , .ResultExps ) == 0 ;  assert Verify ( 3 , .ResultExps ) == 18 ;  .Statements }
          Verify |-> method Verify ( x : int , .ArgDecls ) returns ( r : int , .ArgDecls ) requires x > 0 ensures r > 0 { r := 2 * x ;  r := r * x ;  .Statements }
        </methods>
        <stack>
          .List
        </stack>
        <store>
          0 |-> ( _0:Int , var )
          1 |-> ( 2 *Int _0 *Int _0 , var )
        </store>
        <env>
          r |-> 1
          x |-> 0
        </env>
        <nextLoc>
          2
        </nextLoc>
      </generatedTop>
    }
  #And
    {
      true
    #Equals
      2 *Int _0 *Int _0 >Int 0
    }
  #And
    {
      true
    #Equals
      _0 >Int 0
    }
#Or
    {
      Result
    #Equals
      <generatedTop>
        <k>
          .
        </k>
        <methods>
          Main |-> method Main ( .ArgDecls ) returns ( r : int , .ArgDecls ) requires true ensures true { assert Verify ( 0 , .ResultExps ) == 0 ;  assert Verify ( 3 , .ResultExps ) == 18 ;  .Statements }
          Verify |-> method Verify ( x : int , .ArgDecls ) returns ( r : int , .ArgDecls ) requires x > 0 ensures r > 0 { r := 2 * x ;  r := r * x ;  .Statements }
        </methods>
        <stack>
          .List
        </stack>
        <store>
          0 |-> ( _0:Int , var )
          1 |-> ( _00:Int , var )
        </store>
        <env>
          r |-> 1
          x |-> 0
        </env>
        <nextLoc>
          2
        </nextLoc>
      </generatedTop>
    }
  #And
    {
      false
    #Equals
      _0 >Int 0
    }
